sql/src/main/java/com/amazon/opendistroforelasticsearch/sql/sql/parser/AstAggregationBuilder.java

```diff
@@ -137,6 +139,15 @@ private boolean isNonLiteral(UnresolvedExpression expr) {
     return !(expr instanceof Literal);
   }
 
+  private boolean isNonLiteralFunction(UnresolvedExpression expr) {
+    if (expr instanceof Function) {
+      List<? extends Node> children = expr.getChild();
+      return children.stream()
+              .allMatch(child -> isNonLiteral((UnresolvedExpression) child));
```

Comment:
I think we need to check this recursively, ex. `ABS(ABS(age))`. Could we merge `isNonLiteral()` to this method as base case for recursion?

Reply:
right, I will add.

Reply:
added this change.